File: perl/pi.pl
use strict;
use warnings;
use Time::HiRes;

my $n = 1;
my $sigma = 0.0;
my $pi = 0.0;
my $piConstant = 3.14159265358979323846;
my $digits = 5;
my $constant = 10**$digits;
my $equal = int($piConstant*$constant + 0.5);
my $start = Time::HiRes::gettimeofday();

while(int($pi*$constant+0.5)!=$equal){
  $sigma = $sigma + 1/($n * $n);
  $pi = sqrt($sigma * 6.0);
  $n++;
}

my $end = Time::HiRes::gettimeofday();
printf("%.4f\n", $end - $start);

File: php/pi.php
<?php

function microtime_float()
{
    list($usec, $sec) = explode(" ", microtime());
    return ((float)$usec + (float)$sec);
}

$n=1;
$sigma=0.0;
$pi=0.0;
$piconstant=3.14159265358979323846;
$digits=5;
$constant=10**$digits;
$time_start = microtime_float();

while((int)($pi*$constant+0.5)!=(int)($piconstant*$constant+0.5)){
    $sigma = $sigma + 1/($n*$n);
    $pi = sqrt(6.0 * $sigma);
    $n=$n+1;
}

$time_end = microtime_float();
$time = $time_end - $time_start;
$version = phpversion();
print "Calculated Pi correctly up to $digits digits in $time seconds using PHP $version\n";
?>
